--- setup.py
+++ setup.py
@@ -50,5 +49,0 @@
-    else:
-        if CUDA_HOME is None:
-            raise NotImplementedError('CUDA_HOME is None. Please set environment variable CUDA_HOME.')
-        else:
-            raise NotImplementedError('No CUDA runtime is found. Please set FORCE_CUDA=1 or test it by running torch.cuda.is_available().')
--- src/cpu/ms_deform_attn_cpu.cpp
+++ src/cpu/ms_deform_attn_cpu.cpp
@@ -19 +18,0 @@
-#include <ATen/cuda/CUDAContext.h>
